### PR TITLE
Make the admin api more rest-y.

### PR DESCRIPTION
**Pull request by calavera**
The current scheme is a little messy. This makes it more understandable.

Signed-off-by: David Calavera <david.calavera@gmail.com>

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -95,11 +95,15 @@ func NewAPIWithVersion(ctx context.Context, globalConfig *conf.GlobalConfigurati
 			r.Use(api.requireAdminCredentials)
 
 			r.Get("/users", api.adminUsers)
+			r.Post("/users", api.adminUserCreate)
 
-			r.Post("/user", api.adminUserCreate)
-			r.Get("/user", api.adminUserGet)
-			r.Put("/user", api.adminUserUpdate)
-			r.Delete("/user", api.adminUserDelete)
+			r.Route("/user/{user_id}", func(r *router) {
```
Comment (brycekahle) [inline api/api.go]: Should this be `/users/{user_id}`?

File: cmd/admin_cmd.go [hunk 2]
```diff
@@ -115,7 +115,7 @@ func adminDeleteUser(globalConfig *conf.GlobalConfiguration, config *conf.Config
 
 	user, err := db.FindUserByEmailAndAudience(instanceID, args[0], getAudience(config))
 	if err != nil {
-		user, err = db.FindUserByID(args[0])
+		user, err = db.FindUserByInstanceIDAndID(instanceID, args[0])
```
Comment (brycekahle) [inline cmd/admin_cmd.go]: User IDs are globally unique, which is why I didn't have the `instance_id` here.

File: api/admin_test.go [hunk 3]
```diff
@@ -86,7 +86,7 @@ func (ts *AdminTestSuite) TestAdminUsers() {
 	require.Equal(ts.T(), http.StatusOK, w.Code)
 
 	assert.Equal(ts.T(), "</admin/users?page=1>; rel=\"last\"", w.HeaderMap.Get("Link"))
-	assert.Equal(ts.T(), "2", w.HeaderMap.Get("X-Total-Count"))
+	assert.Equal(ts.T(), "5", w.HeaderMap.Get("X-Total-Count"))
```
Comment (brycekahle) [inline api/admin_test.go]: Any reason for changing the total counts? It is causing the tests to fail now.

File: cmd/admin_cmd.go (hunk 2 above)
Comment (calavera) [reply cmd/admin_cmd.go]: I understand that, but by not checking the InstanceID you are giving admins in other instances access to any user.

File: api/api.go (hunk 1 above)
Comment (calavera) [reply api/api.go]: changed.

File: cmd/admin_cmd.go (hunk 2 above)
Comment (brycekahle) [reply cmd/admin_cmd.go]: Ah that makes sense. 👍

**Pull request merged**